workspaces/src/network/transaction.rs

```diff
@@ -0,0 +1,323 @@
+use crate::worker::Worker;
+use crate::Account;
+
+/// A set of arguments we can provide to a transaction, containing
+/// the function name, arguments, the amount of gas to use and deposit.
+#[derive(Debug, Clone)]
+pub struct Function {
+    name: String,
+    args: Vec<u8>,
+    deposit: Balance,
+    gas: Gas,
+}
+
+impl Function {
+    pub fn new(name: &str) -> Self {
+        Self {
+            name: name.into(),
+            args: vec![],
+            deposit: DEFAULT_CALL_DEPOSIT,
+            gas: DEFAULT_CALL_FN_GAS,
+        }
+    }
+
+    pub fn args(mut self, args: Vec<u8>) -> Self {
+        self.args = args;
+        self
+    }
+
+    pub fn args_json<U: serde::Serialize>(mut self, args: U) -> anyhow::Result<Self> {
+        self.args = serde_json::to_vec(&args)?;
+        Ok(self)
+    }
+
+    pub fn args_borsh<U: borsh::BorshSerialize>(mut self, args: U) -> anyhow::Result<Self> {
+        self.args = args.try_to_vec()?;
+        Ok(self)
+    }
+
+    pub fn deposit(mut self, deposit: u128) -> Self {
+        self.deposit = deposit;
+        self
+    }
+
+    pub fn gas(mut self, gas: u64) -> Self {
+        self.gas = gas;
+        self
+    }
+}
+
+impl From<Function> for Action {
+    fn from(function: Function) -> Self {
+        Self::FunctionCall(FunctionCallAction {
+            method_name: function.name,
+            args: function.args,
+            deposit: function.deposit,
+            gas: function.gas,
+        })
+    }
+}
+
+pub struct Transaction<'a> {
+    client: &'a Client,
+    signer: InMemorySigner,
+    receiver_id: AccountId,
+    actions: Vec<Action>,
+}
+
+impl<'a> Transaction<'a> {
+    pub(crate) fn new(client: &'a Client, signer: InMemorySigner, receiver_id: AccountId) -> Self {
+        Self {
+            client,
+            signer,
+            receiver_id,
+            actions: Vec::new(),
+        }
+    }
+
+    /// Adds a key to the `receiver_id`'s account, where the public key can be used
+    /// later to delete the same key.
+    pub fn add_key(mut self, pk: PublicKey, ak: AccessKey) -> Self {
+        self.actions.push(
+            AddKeyAction {
+                public_key: pk.into(),
+                access_key: ak.into(),
+            }
+            .into(),
+        );
+        self
+    }
+
+    /// Call into the `receiver_id`'s contract with the specific function arguments.
+    pub fn call(mut self, function: Function) -> Self {
+        self.actions.push(function.into());
+        self
+    }
+
+    /// Create a new account with the account id being `receiver_id`.
+    pub fn create_account(mut self) -> Self {
+        self.actions.push(CreateAccountAction {}.into());
+        self
+    }
+
+    /// Deletes the `receiver_id`'s account. The beneficiary specified by
+    /// `beneficiary_id` will receive the funds of the account deleted.
+    pub fn delete_account(mut self, beneficiary_id: AccountId) -> Self {
```

Comment:
nothing potentially to change now, but these parameters might in the future be a reference to the `AccountId`, or is that not possible here for some reason? Asking to potentially avoid a breaking change down the road.

Reply:
good catch, a lot of this was generated code from github copilot so didn't catch this at first glance or the type aliases you mentioned in another comment